src/net/mod.rs

```diff
@@ -106,6 +106,12 @@ pub async fn init(ct: CancellationToken) -> Result<(), Error> {
 		NS.clone(),
 		DB.clone(),
 		ID.clone(),
+		AUTH_NS.clone(),
```

Comment:
What's the difference between `NS`, `AUTH_NS`, and `NS_LEGACY`?

Reply:
The `NS_LEGACY` header is the old `NS` header, which uses the the key "NS". It was was replaced in #2985 by the current `NS` header, which uses the key "surreal-ns". Both headers are used to specify the namespace in which to run the query that is sent to the HTTP REST API.

In that same PR, we introduced the `AUTH_NS` and `AUTH_DB`, which are used to specify level at which to perform authentication. A request can be performed against a database (thus requiring the `NS` and `DB`  headers to be specified) while authentication is performed at the namespace level via the `AUTH_NS` header.